### PR TITLE
Week 3 PR

### PR DESCRIPTION
**Pull request by rwackes**
In this PR, I continued to work on my comments page to make the following changes: 
- I added a button and the functionality to delete every single user-submitted comment currently being stored in Datastore. Screenshot: https://screenshot.googleplex.com/KxPQLzz7EXj
- I allow the user to select a number of comments to load at a time on the webpage, which is especially helpful once my page accumulates many more comments. The default number of comments that will load are the five most recently submitted comments. Screenshot: https://screenshot.googleplex.com/iAGFK5YmFmJ

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
```
Comment (vkkpng-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:23]: Alphabetize the imports (this is in the style guide internally so probably not know to interns)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -63,9 +65,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       long timestamp = (long) entity.getProperty("timestamp");
       String message = (String) entity.getProperty("message");
       FormSubmissions comment = new FormSubmissions(id, fname, lname, email, number, timestamp, message);
+
       comments.add(comment);
-      //commentCount -= 1;
```
Comment (vkkpng-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:67]: You can remove this. 

Instead in L57, check if length of  List<FormSubmissions> comments  == commentsCount

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -63,9 +65,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       long timestamp = (long) entity.getProperty("timestamp");
       String message = (String) entity.getProperty("message");
       FormSubmissions comment = new FormSubmissions(id, fname, lname, email, number, timestamp, message);
```
Comment (vkkpng-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:67]: In a design perspective, when should we post comments on the website. What if someone doesn't actually enter anything, but press the submit button? What if they don't write a message?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -63,9 +65,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+
       comments.add(comment);
-      //commentCount -= 1;
+      commentCount -= 1;
     }
+
     response.setContentType("application/json");
     String commentsJSON = convertToJsonUsingGson(comments);
     response.getWriter().println(commentsJSON);
```
Comment (vkkpng-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:75]: In production code we don't include print statements (no need to remove) but we will enforce this in the capstone project

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -109,7 +95,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     long timestamp = System.currentTimeMillis();
```
Comment (vkkpng-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:95]: I couldn't leave a comment on the code above, but i really like these comments!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -109,7 +95,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
     // Get the message input from the form.
     String message = getParameter(request, "message", "");
-
+    
     Entity formSubmissionEntity = new Entity("FormSubmissions");
```
Comment (vkkpng-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:99]: In a design perspective, when should we post comments on the website. What if someone doesn't actually enter anything, but press the submit button? What if they don't write a message?

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -12,49 +12,27 @@
...
+    })
   });
 }
 
-/** Creates an <li> element containing form submission/comment. */
+/** 
+ * Creates an <li> element containing a form submission/comment.
+ */
 function createCommentElement(comment) {
   const commentElement = document.createElement('li');
```
Comment (vkkpng-g) [inline portfolio/src/main/webapp/script.js:37]: Next todo: add a comment what this parameter comment.

I would perhaps also change the naming to 

server_comment (or something distinguishable). I was at first confused where this variable was coming from in L42)

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -75,6 +58,7 @@ function createCommentElement(comment) {
   });
   
   commentElement.appendChild(titleElement);
```
Comment (vkkpng-g) [inline portfolio/src/main/webapp/script.js:60]: Very good code structure where you defined the variable first and then used them in below. Also noticed you defined them in the order they are presented which is also great.

Comment (wdi-gg) [reply portfolio/src/main/webapp/script.js:60]: +1. Very clean and easy to read!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (wdi-gg) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:99]: +1. Validate input on client side.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (wdi-gg) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:67]: I deleted my comments since we are making the same point here.

**Review by wdi-gg — COMMENTED** [review]
It's a bit confusing to review code from 2 pull request, but basic functionality works well!

**Review by vkkpng-g — COMMENTED** [review]
Nice!

**Review by vkkpng-g — APPROVED** [review]

**Pull request merged**